readme.md

```diff
@@ -1,161 +1,181 @@
+- [Technical requirements](#technical-requirements)
+- [Specification requirements](#specification-requirements)
+- [Supported protocols](#supported-protocols)
+- [How to use the template](#how-to-use-the-template)
+  * [CLI](#cli)
+  * [Docker](#docker)
+- [Template configuration](#template-configuration)
+- [Custom hooks that you can disable](#custom-hooks-that-you-can-disable)
+- [Development](#development)
+- [Contributors](#contributors)
+
+<!-- tocstop -->
+
+## Overview
+
+<!--  
+The overview should explain in just a few sentences the template's purpose and its most essential features.
+-->
+
+This template generates a Go module that uses [watermill](https://github.com/ThreeDotsLabs/watermill) as the messaging middleware
+
+## Technical requirements
+
+<!--  
+Specify what version of the Generator is your template compatible with. This information should match the information provided in the template configuration under the `generator` property.
+-->
+
+- 1.1.0 =< [Generator](https://github.com/asyncapi/generator/) < 2.0.0,
+- Generator specific [requirements](https://github.com/asyncapi/generator/#requirements)
+
+## Specification requirements
+
+<!--  
+The template might need some AsyncAPI properties that normally are optional. For example code generator might require some specific binding information for a given protocol. Even though you can provide defaults or fallbacks, you should describe in the readme what is the most optimal set of properties that the user should provide in the AsyncAPI file.
+-->
+
+The table contains information on parts of the specification required by this template to generate the proper output.
+
+Property name | Reason | Fallback | Default
+---|---|---|---
+`components.schemas` | This template supports only schemas that have unique and human-readable names. Such names can also be provided if schemas are described under `components.schemas` and each schema is a separate object with its unique key. | - | -
+
+## Supported protocols
+
+<!--  
+Specify what protocols is your code generator supporting. This information should match the information provided in the template configuration under the `supportedProtocols` property. Don't put this section in your readme if your template doesn't generate code.
+-->
+
+Currently this template supports AMQP subscribers
+
+## How to use the template
+
+<!--  
+Make sure it is easy to try out the template and check what it generates. Instructions for CLI and Docker should be easy to use; just copy/paste to the terminal. In other words, you should always make sure to have ready to use docker-compose set up so the user can quickly check how generated code behaves.
+-->
+
+This template must be used with the AsyncAPI Generator. You can find all available options [here](https://github.com/asyncapi/generator/).
+
+### CLI
+
+As of this initial commit this template has been tested to generate an AMQP subscriber for the following asyncapi.yml file
+
+```yaml
```

Comment:
maybe better put this file in a real file, in test folder, and then your instructions in next sections, to run generator, can point to this file?